azafea/event_processors/endless/country.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2020 - Endless
+#
+# This file is part of Azafea
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+COUNTRIES = {
+    'ABW': 'AW', 'AGO': 'AO', 'ALA': 'AX', 'AND': 'AD', 'ARE': 'AE', 'ARM': 'AM', 'ATA': 'AQ',
+    'ATF': 'TF', 'ATG': 'AG', 'AUT': 'AT', 'BDI': 'BI', 'BEN': 'BJ', 'BES': 'BQ', 'BGD': 'BD',
+    'BHS': 'BS', 'BIH': 'BA', 'BLR': 'BY', 'BLZ': 'BZ', 'BRB': 'BB', 'BRN': 'BN', 'CAF': 'CF',
+    'CHL': 'CL', 'CHN': 'CN', 'COD': 'CD', 'COG': 'CG', 'COK': 'CK', 'COM': 'KM', 'CPV': 'CV',
+    'CUW': 'CW', 'CYM': 'KY', 'DNK': 'DK', 'ESH': 'EH', 'EST': 'EE', 'FLK': 'FK', 'FRO': 'FO',
+    'FSM': 'FM', 'GIN': 'GN', 'GLP': 'GP', 'GNB': 'GW', 'GNQ': 'GQ', 'GRD': 'GD', 'GRL': 'GL',
+    'GUF': 'GF', 'GUY': 'GY', 'IRL': 'IE', 'IRQ': 'IQ', 'ISR': 'IL', 'JAM': 'JM', 'KAZ': 'KZ',
+    'KOR': 'KR', 'LBR': 'LR', 'LBY': 'LY', 'MAC': 'MO', 'MAF': 'MF', 'MDG': 'MG', 'MDV': 'MV',
+    'MEX': 'MX', 'MLT': 'MT', 'MNE': 'ME', 'MNP': 'MP', 'MOZ': 'MZ', 'MTQ': 'MQ', 'MYT': 'YT',
+    'NIU': 'NU', 'PAK': 'PK', 'PCN': 'PN', 'PLW': 'PW', 'PNG': 'PG', 'POL': 'PL', 'PRK': 'KP',
+    'PRT': 'PT', 'PRY': 'PY', 'PYF': 'PF', 'SEN': 'SN', 'SGS': 'GS', 'SLB': 'SB', 'SLV': 'SV',
+    'SPM': 'PM', 'SRB': 'RS', 'SUR': 'SR', 'SVK': 'SK', 'SVN': 'SI', 'SWE': 'SE', 'SWZ': 'SZ',
+    'SYC': 'SC', 'TCD': 'TD', 'TKM': 'TM', 'TUN': 'TN', 'TUR': 'TR', 'TUV': 'TV', 'UKR': 'UA',
+    'URY': 'UY', 'WLF': 'WF'}
+
+
+def tranform_alpha_3_into_2(alpha_3: str) -> str:
+    return COUNTRIES.get(alpha_3, alpha_3[:2])
```

Comment:
The default here makes me very nervous. I think the transformation should fail for unknown country codes, rather than mapping them to some arbitrary pair of letters.

Reply:
It’s not arbitrary. The list of countries has been created from the ISO list, with only the list of 3-letter codes for countries whose 2-letter is different from the two first letters. All the values from the production database have been tested, and one was missing from the ISO list (a temporary code for Kosovo).